client/internal/peer/status.go

```diff
@@ -995,15 +1002,28 @@ func (d *Status) notifyPeerStateChangeListeners(peerID string) {
 	if !ok {
 		return
 	}
+
+	// collect the relevant data for router peers
+	changedPeers := make(map[string]RouterState, len(d.changeNotify))
```

Comment:
am i missing something or we're not considering the peers didnt changed when recalculating routes? `convertRouterPeerStatuses` only gets updated peers and passes to `recalculateRoutes`